### PR TITLE
feat(boundary): Support boundary events for intermediate connectors

### PR DESCRIPTION
**Pull request by sbuettner**
Support boundary events for intermediate process instance based connectors

## Related issues

Relates to https://github.com/camunda/team-connectors/issues/507

### CODE REVIEW

**Review by chillleader — APPROVED** [review]
LGTM

File: connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/DefaultProcessInstanceContext.java [hunk 1]
```diff
@@ -68,7 +65,8 @@ public Long getKey() {
 
   @Override
   public <T> T bind(final Class<T> cls) {
-    HashMap<String, Object> copyOfProperties = new HashMap<>(contextProperties);
+    // TODO we should improve caching to avoid fetching variables for very bind.
```
Comment (igpetrov) [inline connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/DefaultProcessInstanceContext.java:68]: Shall we also have a ticket for the `TODO`?

File: connector-sdk/core/src/main/java/io/camunda/connector/api/inbound/correlation/BoundaryEventCorrelationPoint.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    implements ProcessCorrelationPoint {
+  @Override
+  public String getId() {
+    return null;
+  }
+
+  @Override
+  public int compareTo(ProcessCorrelationPoint o) {
+    return 0;
+  }
```
Comment (igpetrov) [inline connector-sdk/core/src/main/java/io/camunda/connector/api/inbound/correlation/BoundaryEventCorrelationPoint.java:33]: Don't we need this implemented?

**Review by igpetrov — APPROVED** [review]
LGTM, minor thoughts.

File: connector-sdk/core/src/main/java/io/camunda/connector/api/inbound/correlation/BoundaryEventCorrelationPoint.java (hunk 2 above)
Comment (sbuettner) [reply connector-sdk/core/src/main/java/io/camunda/connector/api/inbound/correlation/BoundaryEventCorrelationPoint.java:33]: Good point. Actually we dont make use of it. Will remove it.

File: connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/DefaultProcessInstanceContext.java (hunk 1 above)
Comment (sbuettner) [reply connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/DefaultProcessInstanceContext.java]: @igpetrov Yes, added reference ticket

**Pull request merged**